apps/aecore/src/aec_block_genesis.erl

```diff
@@ -58,19 +62,15 @@ genesis_block() ->
       }.
 
 populated_trees() ->
-    {ok, T0} = aec_trees:all_trees_new(),
-    PresetAccounts = aec_genesis_block_settings:preset_accounts(),
-    Trees = add_preset_accounts(PresetAccounts, T0),
-    Trees.
+    populated_trees(aec_genesis_block_settings:preset_accounts()).
 
-add_preset_accounts([], Trees) ->
-    Trees;
-add_preset_accounts([{PubKey, Amt} | T], Trees0) ->
-    Account = aec_accounts:new(PubKey, Amt, 0),
-    AccountsTrees0 = aec_trees:accounts(Trees0),
-    {ok, AccountsTrees} = aec_accounts:put(Account, AccountsTrees0),
-    Trees = aec_trees:set_accounts(Trees0, AccountsTrees),
-    add_preset_accounts(T, Trees).
+populated_trees(PresetAccounts) ->
+     {ok, Trees0} = aec_trees:all_trees_new(),    %% renaming required in aec_trees!
```

Comment:
Indeed - I am planning of doing this renaming (ADT for state trees) in PT-153588467.

Reply:
Great. If you're on it, also make aec_accounts and aec_trees less like spaghetti, they call each other recursively.